### PR TITLE
HIVE-26763 : Upgraded zookeeper version to latest 3.6.3 and curator to latest 5.2.0

### PR DESCRIPTION
**Pull request by amanraj2520**
### What changes were proposed in this pull request?
Upgraded zookeeper version to latest 3.6.3 and curator to latest 5.2.0

### Why are the changes needed?
To keep zookeeper and curator updated to latest version.

### CODE REVIEW

**Comment by dengzhhu653** [comment]
Hi @amanraj2520, there are some test failures on the latest changes, cloud you please check that?
`java.lang.IllegalStateException: Timed out waiting for watch removal`

File: pom.xml [hunk 1]
```diff
@@ -203,7 +203,7 @@
     <velocity.version>2.3</velocity.version>
     <xerces.version>2.12.2</xerces.version>
     <xmlsec.version>2.3.0</xmlsec.version>
-    <zookeeper.version>3.5.5</zookeeper.version>
+    <zookeeper.version>3.6.3</zookeeper.version>
```
Comment (dengzhhu653) [inline pom.xml:206]: we should also upgrade the zookeeper in standable-metastore/pom.xml

**Comment by amanraj2520** [comment]
@dengzhhu653 @zabetak Can you please review these changes.

**Comment by dengzhhu653** [comment]
The changes overall look good to me, and I tested against Zookeeper 3.5.5 and was able to get HiveServer2/Metastore uris, though with a warn message:

[main-SendThread(zookeeper-1:2181)]: WARN zookeeper.ClientCnxn: An exception was thrown while closing send thread for session 0x10b8b7297f602c3.
org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x10b8b7297f602c3, likely server has closed socket
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77) ~[zookeeper-3.6.3.jar:3.6.3]
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350) ~[zookeeper-3.6.3.jar:3.6.3]
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1290) ~[zookeeper-3.6.3.jar:3.6.3]

**Review by dengzhhu653 — APPROVED** [review]

**Comment by amanraj2520** [comment]
@dengzhhu653 Can you please merge this PR

**Comment by dengzhhu653** [comment]
> @dengzhhu653 Can you please merge this PR

I would like to leave it open for two work days, see if there are more comments from other reviewers, otherwise I will merge it. Thank you for your contribution!

**Review by zabetak — APPROVED** [review]

**Pull request merged**